spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -691,12 +691,18 @@ private void initPartitionsIfNeeded() {
 
 					if (offset < 0) {
 						this.consumer.seekToEnd(topicPartition);
-						newOffset = this.consumer.position(topicPartition) + offset;
+						newOffset = Math.max(0, this.consumer.position(topicPartition) + offset);
 					}
 
-					this.consumer.seek(topicPartition, newOffset);
-					if (this.logger.isDebugEnabled()) {
-						this.logger.debug("Reset " + topicPartition + " to offset " + newOffset);
+					try {
+						this.consumer.seek(topicPartition, newOffset);
+						if (this.logger.isDebugEnabled()) {
+							this.logger.debug("Reset " + topicPartition + " to offset " + newOffset);
+						}
+					}
+					catch (Exception e) {
+						logger.error("Failed to set initial offset for " + topicPartition
+								+ " at " + newOffset, e);
```

Comment:
I think it's ok to reposition from the results of the seek; the most likely is new records have been added - if someone else has done a seek with the same group-id - it's up to them,

I suppose we could add the position that to the log but we should capture it in a variable rather than calling it again.

Do you want to do during the merge, or do you want me to do it?